lib/rules/no-unused-class-component-methods.js

```diff
@@ -0,0 +1,152 @@
+/**
+ * @fileoverview Prevent declaring unused methods of component class
+ * @author Paweł Nowak
+ */
+
+'use strict';
+
+const Components = require('../util/Components');
+const astUtil = require('../util/ast');
+const docsUrl = require('../util/docsUrl');
+
+// ------------------------------------------------------------------------------
+// Rule Definition
+// ------------------------------------------------------------------------------
+
+const internalMethods = [
```

Comment:
what about unused static methods?

Reply:
Should we be concerned about static methods? They doesn't seem to be closely connected to component. I feel that static methods could be used outside of component and in that case it would be harder to determine if method is used..

Reply:
In a react component, yes - it's not meant to be used as a class.

(An instance method could also be used outside of the component)

Perhaps maybe an option to control checking of static methods?

Reply:
That might be a good idea. I'll try to tackle it after basic version.